### PR TITLE
Removed headers and messageID

### PR DESCRIPTION
**Pull request by rconjour**
Removed php based messageID, this gives in outlook.com errors and removed headers. So elgg emails wouldn't be marked as spam anymore.

### CODE REVIEW

File: engine/lib/notification.php [hunk 1]
```diff
@@ -645,9 +645,6 @@ function elgg_send_email($from, $to, $subject, $body, array $params = null) {
 	$message->setSubject($subject);
 	$message->setBody($body);
 
-	foreach ($result['headers'] as $headerName => $headerValue) {
```
Comment (jeabakker) [inline engine/lib/notification.php]: this should not be removed, as plugin hooks can return header lines which need to be set

**Review by jeabakker — CHANGES_REQUESTED** [review]
Why do you believe this will help in spam prevention? Any source article/knowledge base article that supports the statement?

**Comment by rconjour** [comment]
If you check at https://www.mail-tester.com you can see that spam assassin fails on the Message ID. And by trail and error we've found out that outlook.com doesn't accept the emails untill we removed the headers and left it to postfix or any other mail handler (f.e dovecot).

The current header didn't approve to the RFC-2822 protocol.

**Comment by jeabakker** [comment]
do you have any examples of failed `message-id` headers so we can maybe improve the headers

**Comment by rconjour** [comment]
The email has this messageID:
Message-ID: =?UTF-8?Q?</.default.1500386798.1914@domainremovedbyme.nl>?=

After the changes it looks like this:
Message-ID: <20170718145925.AB2F460138D6@domainremovedbyme.nl>

**Comment by jeabakker** [comment]
i think i found the problem, as you can see the failed message-id is UTF-8 encoded, the other isn't.

i'm trying to find out why this is happening

**Comment by cleverthings** [comment]
Great job rconjour & jeabakker for catching that one and for your persistence in fixing it - you are both awesome!

**Comment by hypeJunction** [comment]
I've had issues with Outlook due to wrong content-type headers but not message id. I don't really see why message id would qualify something as spam, though incorrect UTF encoding might be the culprit.

**Comment by hypeJunction** [comment]
I'd dig through Zend Mail code to see when it forces UTF encoding. Perhaps the ID contains multi-byte characters, what's the actual domain you use?

**Comment by jeabakker** [comment]
I found the issue and will create a new PR to fix it, closing this PR

**Pull request closed without merging**